### PR TITLE
feat: thank you page

### PR DESCRIPTION
**Pull request by olehrakwix**
result UI:
image

inspired by this UI:
image

### CODE REVIEW

**Review by alisaLachman — APPROVED** [review]

**Comment by VladyslavG** [comment]
how about mobile screen size ?

File: app/routes/thank-you/route.tsx [hunk 1]
```diff
@@ -13,14 +17,29 @@ export default function ThankYouPage() {
...
         <div className={styles.root}>
             <div className={styles.text}>
                 <h1 className={styles.title}>Thank You!</h1>
-                <p className={styles.paragraph}>
-                    You will receive a confirmation email soon. Your order number: {orderId}
-                </p>
+                <div className={styles.paragraph}>
+                    <div>You will receive a confirmation email soon.</div>
+                    <br />
+                    {order ? <div>Your order number: {order.number}</div> : null}
```
Comment (yurii-ve) [inline app/routes/thank-you/route.tsx]: `{order && <div>...</div>}`?

File: app/routes/thank-you/route.tsx [hunk 2]
```diff
@@ -13,14 +17,29 @@ export default function ThankYouPage() {
...
     return (
         <div className={styles.root}>
             <div className={styles.text}>
                 <h1 className={styles.title}>Thank You!</h1>
-                <p className={styles.paragraph}>
-                    You will receive a confirmation email soon. Your order number: {orderId}
-                </p>
+                <div className={styles.paragraph}>
+                    <div>You will receive a confirmation email soon.</div>
+                    <br />
```
Comment (yurii-ve) [inline app/routes/thank-you/route.tsx]: why not add some gap with css?

File: src/components/order-summary/line-item/line-item.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+export interface ImageUrlOptions {
+    fitWidth: number;
+    fitHeight: number;
+    quality?: number;
+}
+
+const DEFAULT_IMAGE_QUALITY = 90;
+
+export function getImageUrl(imageUrl: string, options?: ImageUrlOptions) {
```
Comment (yurii-ve) [inline src/components/order-summary/line-item/line-item.tsx]: let's move it out of the component file.. put in the `utils.ts` in the component folder

File: src/components/order-summary/line-item/line-item.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
+import { orders } from '@wix/ecom';
+import styles from './line-item.module.scss';
+
+export interface LineItemProps {
+    item: orders.OrderLineItem;
+}
+
+export const LineItem = ({ item }: LineItemProps) => {
+    const lineItemImage = item.image
```
Comment (yurii-ve) [inline src/components/order-summary/line-item/line-item.tsx]: `lineItemImageUrl`

File: src/components/order-summary/line-item/line-item.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+              fitHeight: 200,
+              fitWidth: 200,
+          })
+        : null;
+
+    return (
+        <div className={styles.root}>
+            <div className={styles.productInfo}>
+                <div>
+                    {lineItemImage ? (
```
Comment (yurii-ve) [inline src/components/order-summary/line-item/line-item.tsx]: `{lineItemImage && ...}`

File: src/components/order-summary/line-item/line-item.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+                        <img src={lineItemImage} alt={item.productName?.translated ?? ''} />
+                    ) : null}
+                </div>
+
+                <div className={styles.productDetails}>
+                    <div className={styles.productName}>{item.productName?.translated}</div>
+                    <div>{item.price?.formattedAmount}</div>
+                    {item.descriptionLines?.map((l) => (
```
Comment (yurii-ve) [inline src/components/order-summary/line-item/line-item.tsx]: `l` => `descriptionLine`

File: _codux/boards/components/order-summery/order-summery.board.tsx [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import { createBoard } from '@wixc3/react-board';
+import { OrderSummary } from '../../../../src/components/order-summary/order-summary';
+
+export default createBoard({
+    name: 'OrderSummary',
+    Board: () => <OrderSummary order={{}} />,
```
Comment (yurii-ve) [inline _codux/boards/components/order-summery/order-summery.board.tsx]: why does this board takes an empty object order?

File: src/components/order-summary/order-summary.tsx [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    return (
+        <div className={styles.root}>
+            <div className={cx(styles.section)}>
+                <div className={styles.lineItems}>
+                    {order.lineItems?.map((i) => (
+                        <LineItem key={i._id} item={i} />
+                    ))}
+                </div>
+
+                <hr className={styles.divider} />
```
Comment (yurii-ve) [inline src/components/order-summary/order-summary.tsx]: `hr` doesn't fit here
```
The <hr> HTML element represents a thematic break between paragraph-level elements: for example, a change of scene in a story, or a shift of topic within a section.
```

File: src/components/order-summary/order-summary.tsx [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+ * This component was created using Codux's Default new component template.
+ * To create custom component templates, see https://help.codux.com/kb/en/article/kb16522
+ */
+
+export const OrderSummary = ({ order }: OrderSummaryProps) => {
+    return (
+        <div className={styles.root}>
+            <div className={cx(styles.section)}>
+                <div className={styles.lineItems}>
+                    {order.lineItems?.map((i) => (
```
Comment (yurii-ve) [inline src/components/order-summary/order-summary.tsx]: `i` -> `lintItem`

File: src/components/order-summary/order-summary.tsx [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+                        <LineItem key={i._id} item={i} />
+                    ))}
+                </div>
+
+                <hr className={styles.divider} />
+
+                <div className={styles.priceDetailsWrapper}>
+                    <div />
```
Comment (yurii-ve) [inline src/components/order-summary/order-summary.tsx]: ?

File: src/components/order-summary/order-summary.tsx [hunk 11]
```diff
@@ -0,0 +1,112 @@
...
+                    </div>
+                </div>
+            </div>
+
+            <div className={cx(styles.section, styles.deliveryBilling)}>
+                <div className={styles.delivery}>
+                    <h6 className={styles.title}>Delivery address</h6>
+                    <ul className={styles.addressLines}>
+                        <li>
+                            {`${order?.shippingInfo?.logistics?.shippingDestination?.contactDetails?.firstName} ${order?.shippingInfo?.logistics?.shippingDestination?.contactDetails?.lastName}`}
```
Comment (yurii-ve) [inline src/components/order-summary/order-summary.tsx]: it's not readable.. let's destructure `firstName` and `lastName` somewhere in the component body

File: src/components/order-summary/order-summary.tsx [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+                        {renderAddress(order?.billingInfo?.address)}
+                        <li>{order?.billingInfo?.contactDetails?.phone}</li>
+                    </ul>
+                </div>
+            </div>
+        </div>
+    );
+};
+
+function renderAddress(address: orders.Address | undefined) {
```
Comment (yurii-ve) [inline src/components/order-summary/order-summary.tsx]: let's do this in the component body

File: src/components/order-summary/order-summary.tsx [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+import cx from 'classnames';
+import { LineItem } from './line-item/line-item';
+import styles from './order-summary.module.scss';
+
+export interface OrderSummaryProps {
+    order: orders.Order;
+}
+
+/**
+ * This component was created using Codux's Default new component template.
```
Comment (yurii-ve) [inline src/components/order-summary/order-summary.tsx]: remove this comment

**Comment by alisaLachman** [comment]
@olehrakwix 

looks great, just one small adjustment, align the prices more to the right:

Before:
Screenshot 2024-09-10 at 12 18 28
After:
Screenshot 2024-09-10 at 12 21 07
Screenshot 2024-09-10 at 12 21 18



@ShireeWill 
is anything else is missing regarding the information in the thank you page?

**Comment by olehrakwix** [comment]
@alisaLachman , addressed comments, please see updated view in description.
regarding buyer note: if API will return it - it will be shown, but I have never seen it in real flow

**Review by yurii-ve — APPROVED** [review]

**Pull request merged**